docs: update realtime websockets guide (#19660)
directus/directus: docs/guides/real-time/getting-started/websockets.md

---
contributors: Kevin Lewis, Esther Agbaje
description: "Learn how to get started with Directus' WebSockets interface."
---

> # Getting Started With WebSockets (with the SDK)

Reviewer: ```suggestion
# Getting Started with WebSockets and the Directus SDK
```

You can connect to a Directus project using a WebSocket interface and get updates on data held in a collection in
real-time.

This guide will show you how to get started with Directus' Realtime SDK and JavaScript. WebSockets are
language-agnostic, so you can apply the same set of steps in your stack of choice.

> ::: info WebSockets without the SDK
> 
> To get started with WebSockets without the SDK please refer to this guide
> [here](/guides/real-time/getting-started/websockets-js.md).
> 
> :::

Reviewer: ```suggestion
::: info Want to use WebSockets without the SDK?

To get started with WebSockets without the SDK please refer to our
[Getting Started with WebSockets guide](/guides/real-time/getting-started/websockets-js.md).

:::
```

Make sure to replace `your-directus-url` and `your-access-token` with your project and user details.

## Import the SDK Composables

At the top of the `<script>` tag, import the composables needed for the SDK

```html
<!DOCTYPE html>
<html>
	<body>
		<script>

>       import { createDirectus, staticToken, realtime } from "https://www.unpkg.com/@directus/sdk@11.0.3/dist/ index.js" // [!code ++]

Reviewer: ```suggestion
      import { createDirectus, staticToken, realtime } from "https://www.unpkg.com/@directus/sdk/dist/index.js" // [!code ++]
```

   application technology stack.
2. To verify that the connection is still active.

> In order to prevent the connection from closing, you may reply with a pong event:

Reviewer: No idea why it was italicized before - good catch

<html>
	<body>
		<script>

      import { createDirectus, staticToken, realtime } from "https://www.unpkg.com/@directus/sdk@11.0.3/dist/index.js"

      const url = 'wss://your-directus-url/websocket';
      const access_token = 'your-access-token';
      const collection = 'messages';

      const client = createDirectus(url)
        .with(staticToken(access_token))
        .with(realtime());

>       await client.connect();

Reviewer 2: This is not wrong but also not technically required 😄 `client.subscribe` will also attempt to connect to the websocket if a connection has not yet been established

Reviewer 2: explicitly opening the connection seems a lot less error prone than relying on the `subscribe` function to do this so marking this as resolved
